pkg/test/step.go

```diff
@@ -280,7 +285,11 @@ func (s *Step) CheckResource(expected runtime.Object, namespace string) []error
 
 		actuals = append(actuals, actual)
 	} else {
-		actuals, err = list(cl, gvk, namespace)
+		m, err := meta.Accessor(expected)
+		if err != nil {
+			return append(testErrors, err)
+		}
+		actuals, err = list(cl, gvk, namespace, m.GetLabels())
```

Comment:
Thanks!

Looks like linter caught a logic error here, though:

```
pkg/test/step.go:292:12: ineffectual assignment to err (ineffassign)
		actuals, err = list(cl, gvk, namespace, m.GetLabels())
		         ^
make: *** [Makefile:39: lint] Error 1
```

From a quick glance it seems that introducing `:=` [on line 288](https://github.com/kudobuilder/kuttl/pull/488/files#diff-f83daba5f9026600aa38b2b1e5a0c585535dc3355457e4762c50bdb0f6b0febdR288) caused the new `err` to shadow the one on the top level, resulting in the assignment on line 292 to be effectively lost?

Please give it some careful thought. This function is a subtle and critical part of the code, we should avoid introducing a regression here.

Reply:
good catch, sorry about that.